common/src/main/java/io/netty/util/internal/CleanerJava25Linker.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Array;
+import java.nio.ByteBuffer;
+
+import static java.lang.invoke.MethodType.methodType;
+
+public class CleanerJava25Linker implements Cleaner {
+    private static final InternalLogger logger;
+
+    private static final MethodHandle INVOKE_MALLOC;
+    private static final MethodHandle INVOKE_CREATE_BYTEBUFFER;
+    private static final MethodHandle INVOKE_FREE;
+
+    static {
+        boolean suitableJavaVersion;
+        if (System.getProperty("org.graalvm.nativeimage.imagecode") != null) {
+            // native image supports this since 25, but we don't use PlatformDependent0 here, since
+            // we need to initialize CleanerJava25Linker at build time.
+            String v = System.getProperty("java.specification.version");
+            try {
+                suitableJavaVersion = Integer.parseInt(v) >= 25;
+            } catch (NumberFormatException e) {
+                suitableJavaVersion = false;
+            }
+            // also need to prevent initializing the logger at build time
+            logger = null;
+        } else {
+            // Only attempt to use MemorySegments on Java 25 or greater, because of the following JDK bugs:
+            // - https://bugs.openjdk.org/browse/JDK-8357145
+            // - https://bugs.openjdk.org/browse/JDK-8357268
+            suitableJavaVersion = PlatformDependent0.javaVersion() >= 25;
```

Comment:
Just wanted to point out that segments allocated by `MemorySegment.ofAddress()` are living in the global arena which is unaffected by the checks in `DirectBuffer.address()`, so this could be enabled on java 24 as well (not suitable for the arena based allocation as native access has to be enabled to call `MemorySegment.reinterpret()` - if native access is enabled this method is used anyway)